tests/helpers/test_config_entry_oauth2_flow.py

```diff
@@ -595,15 +595,26 @@ async def test_abort_discovered_existing_entries(
     assert result["reason"] == "already_configured"
 
 
+@pytest.mark.parametrize(
+    ("my_enabled", "expected_redirect_uri"),
+    [
+        (False, "https://example.com/auth/external/callback"),
+        (True, "https://my.home-assistant.io/redirect/oauth"),
+    ],
+)
 @pytest.mark.usefixtures("current_request_with_host")
 async def test_full_flow(
     hass: HomeAssistant,
     flow_handler: type[config_entry_oauth2_flow.AbstractOAuth2FlowHandler],
     local_impl: config_entry_oauth2_flow.LocalOAuth2Implementation,
     hass_client_no_auth: ClientSessionGenerator,
     aioclient_mock: AiohttpClientMocker,
+    my_enabled: bool,
+    expected_redirect_uri: str,
 ) -> None:
     """Check full flow."""
+    if my_enabled:
```

Comment:
I can't think of a better solution at the moment. We try to avoid conditions in tests.

Reply:
A workaround to this rule is to use a list of components to enable, then have a for loop setting them up.